src/components/EditCooperationPollForm.vue

```diff
@@ -0,0 +1,287 @@
+  name: 'EditCooperationPollForm',
+  components: {
+    Button,
+    Calendar,
+    Checkbox,
+    Textarea,
+    InputText,
+  },
+  data() {
+    return {
+      pollData: {
+        header: '',
+        description: '',
+        polledHouses: [] as Array<HouseModel>,
+        creationDate: '' as any,
+        completionDate: '' as any,
+      },
+
+      isDisabled: true,
+      isCreationDateHelpActive: false,
+      beginDate: new Date(),
+      finishDate: new Date(),
+
+      v$: useVuelidate(),
+    };
+  },
+  props: {
+    poll: {
+      type: PollModel,
+      required: true,
+    },
+    displayEditPollModal: {
+      type: Boolean,
+      required: true,
+    },
+    showSuccessOperation: {
+      type: Function,
+      required: true,
+    },
+  },
+  validations() {
+    return {
+      pollData: {
+        header: { requiredValidator, pollTitleLenghtValidator, cyrillicLangTitleValidator },
+        description: { requiredValidator, pollDescriptionLenghtValidator, cyrillicLangTitleValidator },
+        polledHouses: { requiredValidator },
+      },
+    };
+  },
+  mounted() {
+    try {
+      this.$store.dispatch(`${StoreModuleEnum.housesStore}/${HousesActionsEnum.SET_HOUSES}`, this.cooperationId);
+      this.initData();
+    } catch {
+      console.log('error was caught during mounting BaseCooperationPoll');
+    }
+  },
+  methods: {
+    async initData() {
+      await this.$store.dispatch(
+        `${StoreModuleEnum.pollsStore}/${PollsActionEnum.SET_SELECTED_POLL}`,
+        this.$props.poll.id
+      );
+
+      this.pollData.header = this.selectedPoll.header ?? '';
+      this.pollData.description = this.selectedPoll?.description ?? 'Повний опис опитування';
+      this.pollData.polledHouses = this.selectedPoll?.polledHouses ?? [];
+      this.pollData.creationDate = this.selectedPoll?.creationDate.toLocaleString('uk-UA');
+      this.pollData.completionDate = this.selectedPoll?.completionDate.toLocaleString('uk-UA');
+
+      this.beginDate = this.selectedPoll?.creationDate;
+      this.finishDate = this.selectedPoll?.completionDate;
+    },
+    async editPoll() {
+      const data = {
+        header: this.pollData.header,
+        description: this.pollData.description,
+        creationDate: new Date(this.beginDate.toLocaleString('en-US')).toISOString(),
+        completionDate: new Date(this.finishDate.toLocaleString('en-US')).toISOString(),
+        status: this.$props.poll.status,
+        polledHouses: this.pollData.polledHouses,
+      };
+
+      const ids = { cooperationId: this.cooperationId, pollId: this.$props.poll.id };
+
+      await this.$store.dispatch(`${StoreModuleEnum.pollsStore}/${PollsActionEnum.UPDATE_POLL}`, {
+        data,
+        ids,
+      });
+
+      this.$props.showSuccessOperation('редаговано');
+      this.$emit('close-edit-poll');
+    },
+    onChangeCreationDate() {
+      const dateTomorrow = new Date(new Date().getFullYear(), new Date().getMonth(), new Date().getDate() + 1);
```

Comment:
i guess there are simpler ways to do it

```suggestion
      const dateTomorrow = new Date();
      dateTomorrow.setDate(dateTomorrow.getDate() + 1)
```
it is not a good solution to create date 4 times in a row

Reply:
Yeah, thanks!